alerter.c

```diff
@@ -11,22 +12,37 @@ int networkAlertStub(float celcius) {
     return 200;
 }
 
-void alertInCelcius(float farenheit) {
+int testNetworkAlertStub(float celcius) {
+    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+    // Return 200 for ok
+    // Return 500 for not-ok
+    // stub always succeeds and returns 200
+    return returnAlertStub;
+}
+
+void alertInCelcius(float farenheit,int (*fun_ptr_NetworkAlert)(float)) {
     float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
+    int returnCode = (*fun_ptr_NetworkAlert)(celcius);

```

Comment:
Good approach to use function pointer to abstract the dependency 👍 
It would be good to keep test code in different file separate from production code !